xknx/core/state_updater.py

```diff
@@ -20,150 +18,139 @@
 
 
 class StateUpdater:
-    """Class for keeping the states of RemoteValues up to date."""
+    """Class for keeping the state of a RemoteValue up to date."""
+
+    state_updater_semaphore = asyncio.Semaphore(value=2)
 
-    def __init__(self, xknx: XKNX, parallel_reads: int = 2):
+    def __init__(
+        self,
+        xknx: XKNX,
+        remote_value: RemoteValue[Any, Any],
+        sync_state: bool | int | float | str = True,
+    ):
         """Initialize StateUpdater class."""
         self.xknx = xknx
         self.started = False
-        self._workers: dict[int, _StateTracker] = {}
-        self._semaphore = asyncio.Semaphore(value=parallel_reads)
+        self.remote_value = remote_value
+        self.sync_state = sync_state
+        self._worker: _StateTracker | None = None
```

Comment:
Return the worker from `self.register_remote_value()` (maybe make this private?) or just type it non-optional. 
Then the checks `if self._worker:` can be omitted because there is always a worker.

On the other hand a RemoteValue without state address may not need a _StateTracker at all - but with current implementation the seem to have one.